service_taxi/src/main/kotlin/org/thechance/service_taxi/data/gateway/DataBaseGatewayImpl.kt

```diff
@@ -0,0 +1,126 @@
+package org.thechance.service_taxi.data.gateway
+
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.Updates
+import org.bson.types.ObjectId
+import org.koin.core.annotation.Single
+import org.litote.kmongo.and
+import org.litote.kmongo.eq
+import org.litote.kmongo.ne
+import org.thechance.service_taxi.api.models.taxi.toCollection
+import org.thechance.service_taxi.api.models.taxi.toEntity
+import org.thechance.service_taxi.api.models.trip.toCollection
+import org.thechance.service_taxi.api.models.trip.toEntity
+import org.thechance.service_taxi.data.DataBaseContainer
+import org.thechance.service_taxi.data.collection.TaxiCollection
+import org.thechance.service_taxi.data.collection.TripCollection
+import org.thechance.service_taxi.data.utils.isNotNull
+import org.thechance.service_taxi.data.utils.isSuccessfullyUpdated
+import org.thechance.service_taxi.data.utils.paginate
+import org.thechance.service_taxi.data.utils.updateNotNullProperties
+import org.thechance.service_taxi.domain.entity.Taxi
+import org.thechance.service_taxi.domain.entity.TaxiUpdateRequest
+import org.thechance.service_taxi.domain.entity.Trip
+import org.thechance.service_taxi.domain.entity.TripUpdateRequest
+import org.thechance.service_taxi.domain.gateway.DataBaseGateway
+
+@Single
+class DataBaseGatewayImpl(private val container: DataBaseContainer) : DataBaseGateway {
+    // region taxi curd
+    override suspend fun addTaxi(taxi: Taxi): Boolean {
+        return container.taxiCollection.insertOne(taxi.toCollection()).wasAcknowledged()
+    }
+
+    override suspend fun getTaxiById(taxiId: String): Taxi? {
+        return container.taxiCollection.findOneById(ObjectId(taxiId))
+            ?.takeIf { !it.isDeleted }?.toEntity()
+    }
+
+    override suspend fun getAllTaxes(page: Int, limit: Int): List<Taxi> {
+        return container.taxiCollection.find(TaxiCollection::isDeleted ne true)
+            .paginate(page, limit).toList().toEntity()
+    }
+
+    override suspend fun deleteTaxi(taxiId: String): Boolean {
+        return container.taxiCollection.updateOneById(
+            id = ObjectId(taxiId),
+            update = Updates.set(TaxiCollection::isDeleted.name, true),
```

Comment:
Simplify it more by using something like:

```suggestion
        return container.taxiCollection.updateOneById(
            id = ObjectId(taxiId),
            update = set(TaxiCollection::isDeleted setTo true),
```